# Parte 2: Adquisición y procesamiento de datos

Este notebook cumple la función de **validación** de los scripts `download_data.py` y `process_data.py`. Con el objetivo es verificar que se han cumplido los requisitos de adquisición de fuentes y la generación del *Dataset integrado*.

### Dataset integrado:
- **Datos vectoriales**: Límites, manzanas censales, infraestructura
- **Datos raster**: DEM, imágenes satelitales (Sentinel-2/Landsat)
- **Datos tabulares**: Censo, socioeconómicos, ambientales
- **Red vial**: Desde OpenStreetMap usando OSMnx
- **Base de datos espacial**: Todo cargado en PostGIS

### Fuentes de datos mínimas requeridas
La siguiente tabla detalla todos los datos requeridos del enunciado, obtenidos para la comuna de San Joaquín en la carpeta de descarga `data/raw`:

| Tipo de Dato | Archivo obtenido (Raw) | Fuente | Uso Principal | 
| :--- | :--- | :--- | :--- | 
| **Límites administrativos** | `comuna_boundaries_oficial.geojson` | IDE Chile (API) | Recorte de capas y definición de zona de estudio | 
| **Manzanas censales** | `manzanas_censales.geojson` | INE (ArcGIS REST) | Unidad mínima de análisis espacial (vectores) |
| **Censo 2017** | `Censo2017_Manzanas.csv` | INE (Redatam) | Variables demográficas y socioeconómicas (tabla) | 
| **Red vial** | `osm_network.graphml` | OpenStreetMap (OSMnx) | Cálculo de accesibilidad, centralidad y densidad | 
| **Uso del suelo** | `uso_suelo_minvu.geojson`  | IDE Minvu | Zonificación y planificación territorial |
| **DEM** | `srtm_dem_32719.tif` | SRTM (NASA/USGS) | Análisis topográfico (pendiente, aspecto) | 
| **Sentinel-2** | `sentinel2_B04.tif` y `B08.tif` | Copernicus (AWS) | Cálculo de índice de vegetación (NDVI) | 

> **Notas:**
> * `Censo2017_Manzanas.csv` está ubicado en: `data/raw/Censo2017_ManzanaEntidad_CSV/...`
> * `uso_suelo_minvu.geojson` fue consolidado desde: `data/raw/uso_suelo_minvu/IPT_Metropolitana/PRC/IPT_13_PRC_San_Joaquin.shp`
> * `srtm_dem_32719.tif` derivado de: `S34W071.hgt` (recortado y reproyectado a EPSG:32719)

### Datos procesados
La siguiente tabla detalla todos los archivos resultantes del pipeline de procesamiento (`process_data.py`), guardados en `data/processed`:

| Categoría | Archivo Procesado | Contenido / Transformación Realizada |
| :--- | :--- | :--- |
| **Dataset integrado** | `manzanas_atributos.geojson` | Unión de geometría de manzanas + Datos Censo (Join por `manzent`). Base del análisis sociodemográfico. |
| | `manzanas_uso_suelo.geojson` | Cruce espacial (Overlay) entre manzanas y zonas de uso de suelo del MINVU. |
| **Métricas calculadas** | `metrics_network.csv` | Métricas avanzadas de red (Betweenness, Densidad vial) por manzana. |
| | `metrics_manzanas.csv` | Métricas básicas por manzana: conteo de edificios y amenidades interiores. |
| **Elementos de red** | `network_nodes_metrics.geojson` | Nodos de la red vial con valores de centralidad calculados. |
| | `osm_network.gpkg` | Grafo vial convertido a formato GeoPackage optimizado para GIS. |
| **Indices raster** | `sentinel2_ndvi.tif` | Índice de Vegetación (NDVI) calculado desde bandas Sentinel-2 (B08 y B04). |
| **Topografía** | `slope.tif` / `aspect.tif` | Mapas de Pendiente (grados) y Orientación derivados del DEM. |
| **Capas base procesadas** | `comuna_boundaries_oficial.geojson` | Límite comunal reproyectado a EPSG:32719 para consistencia espacial. |
| | `manzanas_censales.geojson` | Geometrías de manzanas reproyectadas y con topología limpia. |
| | `uso_suelo_minvu.geojson` | Capa de uso de suelo reproyectada y estandarizada. |
| | `osm_buildings.geojson` | Edificios de OSM filtrados, reproyectados y limpios. |
| | `osm_amenities.geojson` | Puntos de interés (escuelas, hospitales, etc.) reproyectados. |
| **Tablas auxiliares** | `censo_microdatos.csv` | CSV del Censo con codificación corregida y separado por punto y coma. |

## Importante: colocar estos comandos dentro del ambiente virtual de Python, para descargar y procesar los datos:

- `pip install -r requirements.txt`
- `python scripts/download_data.py --comuna "San Joaquín" --year 2024 --sources all`
- `python scripts/process_data.py`

## 1. Verificación de la conexión a la base de datos y la descarga junto al procesamiento de datos.

In [3]:
# Librerías para el manejo de sistema operativo y rutas
import os
from pathlib import Path
# Librerías para el análisis de datos y manejo geoespacial
import pandas as pd
import geopandas as gpd
# Librerías para la conexión a la base de datos PostgreSQL/PostGIS
from sqlalchemy import create_engine, text
# Librería para el manejo de variables de entorno (.env)
from dotenv import load_dotenv

# Cargar variables de entorno
load_dotenv() # Para obtener credenciales de la BD

# --- CONFIGURACIÓN DE RUTAS ---
# Ruta base relativa para navegar entre carpetas
BASE_DIR = Path('..') # Según la raíz del notebook
DATA_RAW = BASE_DIR / 'data' / 'raw'
DATA_PROCESSED = BASE_DIR / 'data' / 'processed'

# --- CONFIGURACIÓN DE CONEXIÓN A POSTGIS ---
# Se obtienen las credenciales o se usan valores por defecto
DB_USER = os.getenv('POSTGRES_USER', 'geouser')
DB_PASS = os.getenv('POSTGRES_PASSWORD', 'geopass')
DB_HOST = os.getenv('POSTGRES_HOST', 'localhost')
DB_PORT = os.getenv('POSTGRES_PORT', '5432')
DB_NAME = os.getenv('POSTGRES_DB', 'geodatabase')

# Crear string de conexión y motor SQLAlchemy
db_url = f"postgresql://{DB_USER}:{DB_PASS}@{DB_HOST}:{DB_PORT}/{DB_NAME}"
engine = create_engine(db_url)

print(f"✔ Éxito -> Conectado a: {DB_NAME} en {DB_HOST}")

✔ Éxito -> Conectado a: geodatabase en postgis


In [5]:
def check_files(directory, file_list):
    """
    Verifica la existencia física de una lista de archivos en el directorio dado
    y reporta su estado (encontrado/perdido) y tamaño.

    Entradas:
        directory (Path): Objeto Path que apunta al directorio a revisar (raw o processed).
        file_list (list): Lista de strings con los nombres exactos de los archivos esperados.

    Salida:
        list: Una lista con los nombres de los archivos que NO fueron encontrados.
    """
    print(f"--- Verificando {directory.name} ---")
    missing = []
    for f in file_list: # Iterar sobre cada archivo esperado
        fpath = directory / f
        if fpath.exists(): # Verificar si el archivo existe físicamente
            # Calcular tamaño en Megabytes para verificar que no esté vacío
            size_mb = fpath.stat().st_size / (1024 * 1024)
            print(f"✔ {f:<30} | {size_mb:.2f} MB")
        else: # En caso de que el archivo no esté
            print(f"ERROR: {f:<30} | NO ENCONTRADO")
            missing.append(f)
    return missing

# --- LISTAS DE VERIFICACIÓN (CHECKLIST) ---
# Basado en la Tabla "Fuentes de Datos Requeridas" del enunciado

# 1. Archivos RAW (datos descargados, sin procesar):
# Lista FINAL basada en los archivos actuales y la tabla de requisitos del enunciado
raw_expected = [
    'manzanas_censales.geojson',       # Manzanas, INE (Unidad de análisis)
    'comuna_boundaries_oficial.geojson', # Límites, IDE Chile
    'osm_buildings.geojson',           # Infraestructura, OpenStreetMap
    'osm_network.graphml',             # Red vial, OpenStreetMap 
    'uso_suelo_minvu.geojson',         # Uso de suelo, IDE Minvu
    'sentinel2_B04.tif',               # Sentinel-2, Copernicus (Banda Roja)
    'sentinel2_B08.tif',               # Sentinel-2, Copernicus (Banda NIR)
    'srtm_dem_32719.tif'               # DEM, SRTM (Topografía reproyectada)
]

# 2. Archivos PROCESSED (Dataset Integrado - Entregable 2)
processed_expected = [
    'manzanas_atributos.geojson',      # Geometría y Censo
    'manzanas_uso_suelo.geojson',      # Geometría y zonificación
    'metrics_network.csv',             # Métricas de Grafos 
    'censo_microdatos.csv',            # CSV limpio y estandarizado
    'network_nodes_metrics.geojson',   # Nodos con centralidad
    'sentinel2_ndvi.tif',              # Índice espectral calculado
    'slope.tif'                        # Pendiente derivada del DEM
]

print("--- Verificación de Archivos ---")
# Ejecutar verificaciones
missing_raw = check_files(DATA_RAW, raw_expected)
print("\n")
missing_proc = check_files(DATA_PROCESSED, processed_expected)

if not missing_raw and not missing_proc: # Si no están ausentes en raw y processed
    print("\n✔ EXCELENTE: Todos los archivos requeridos por la guía están presentes.")
else:
    print("\nATENCIÓN: Faltan archivos solicitados.")

--- Verificación de Archivos ---
--- Verificando raw ---
✔ manzanas_censales.geojson      | 1.88 MB
✔ comuna_boundaries_oficial.geojson | 0.01 MB
✔ osm_buildings.geojson          | 2.93 MB
✔ osm_network.graphml            | 2.26 MB
✔ uso_suelo_minvu.geojson        | 1.15 MB
✔ sentinel2_B04.tif              | 0.30 MB
✔ sentinel2_B08.tif              | 0.30 MB
✔ srtm_dem_32719.tif             | 0.04 MB


--- Verificando processed ---
✔ manzanas_atributos.geojson     | 2.95 MB
✔ manzanas_uso_suelo.geojson     | 6.32 MB
✔ metrics_network.csv            | 0.04 MB
✔ censo_microdatos.csv           | 0.14 MB
✔ network_nodes_metrics.geojson  | 0.43 MB
✔ sentinel2_ndvi.tif             | 0.44 MB
✔ slope.tif                      | 0.01 MB

✔ EXCELENTE: Todos los archivos requeridos por la guía están presentes.


### 1.1. Análisis de los archivos
El proceso de descarga (`download_data.py`) y procesamiento (`process_data.py`) ha concluido exitosamente, verificándose la existencia de los archivos críticos definidos en la guía del laboratorio.

**Observaciones clave:**
- **Vectoriales:** Se cuenta con la cartografía base (límites comunales y manzanas censales) y la infraestructura (edificios y red vial de OpenStreetMap), en donde los tamaños de archivo (como ~6.3MB para uso de suelo) son consistentes con la extensión territorial de San Joaquín.
- **Raster:** Se han generado correctamente los insumos para el análisis ambiental:
    * `sentinel2_ndvi.tif`: Que permitirá evaluar la cobertura vegetal.
    * `slope.tif`: Que es derivado del DEM SRTM, fundamental para analizar la topografía, aunque la comuna de San Joaquín es predominantemente plana.
- **Tabulares:** El archivo `censo_microdatos.csv` y `metrics_network.csv` confirman que se han calculado indicadores numéricos listos para ser unidos espacialmente.

**Conclusión preliminar:** La estructura de directorios cumple con el estándar del proyecto (`data/raw` y `data/processed`), asegurando la reproducibilidad del pipeline.

## 2. Verificación de contenido (Base de Datos)
A continuación, verificamos que los datos no solo existan como archivos, sino que hayan sido **ingestados correctamente en PostGIS** y contengan los atributos necesarios para el análisis exploratorio ESDA.

In [6]:
# --- VALIDACIÓN 1: DATASET INTEGRADO (CENSO + GEOMETRÍA) ---
# Objetivo: Verificar que las manzanas tienen datos de población asociados
# Consulta a la tabla principal generada por process_data.py
sql_manzanas = "SELECT * FROM processed_data.manzanas_atributos LIMIT 5"

try: # Cargar datos usando pandas y la conexión SQLAlchemy
    gdf_manzanas = gpd.read_postgis(sql_manzanas, engine, geom_col='geometry')
    print("✔ Tabla 'processed_data.manzanas_atributos' cargada correctamente.")
    print(f"Columnas disponibles: {list(gdf_manzanas.columns)[:10]} ...")
    display(gdf_manzanas.head(3))
except Exception as e: # En caso de error leyendo las manzanas
    print(f"ERROR: Error leyendo manzanas_atributos: {e}")

✔ Tabla 'processed_data.manzanas_atributos' cargada correctamente.
Columnas disponibles: ['fid', 'cut', 'region_x', 'provincia_x', 'comuna_x', 'nombre_distrito', 'codigo_distrito', 'manzent', 'zona_censal', 'manzana'] ...


,fid,cut,region_x,provincia_x,comuna_x,nombre_distrito,codigo_distrito,manzent,zona_censal,manzana,...,matirrec,p05_1,p05_2,p05_3,p05_4,region_15r,provincia_15r,comuna_15r,id_manzent_15r,geometry
0,31530,13129,REGIÓN METROPOLITANA DE SANTIAGO,SANTIAGO,SAN JOAQUÍN,SAN JOAQUÍN,1,13129011002003,2,3,...,0,40,0,0,0,13,131,13129,13129011002003,"POLYGON ((348139.376 6294680.091, 348137.924 6..."
1,31531,13129,REGIÓN METROPOLITANA DE SANTIAGO,SANTIAGO,SAN JOAQUÍN,SAN JOAQUÍN,1,13129011003009,3,9,...,0,59,0,0,0,13,131,13129,13129011003009,"POLYGON ((348910.060 6294546.980, 348870.420 6..."
2,31532,13129,REGIÓN METROPOLITANA DE SANTIAGO,SANTIAGO,SAN JOAQUÍN,SAN JOAQUÍN,1,13129011003012,3,12,...,0,43,0,0,0,13,131,13129,13129011003012,"POLYGON ((348922.460 6294551.400, 348920.270 6..."


In [7]:
# --- VALIDACIÓN 2: PLANIFICACIÓN TERRITORIAL (USO DE SUELO, MÉTRICAS DE RED) ---
# Para verificar que se ha realizado la intersección espacial con zonas del MINVU.


# 1. Cargamos TODA la tabla (quitamos el LIMIT 5)
sql_network = "SELECT * FROM processed_data.metrics_network"

try:
    df_net = pd.read_sql(sql_network, engine)
    print(f"✔ Tabla cargada exitosamente. Total de manzanas procesadas: {len(df_net)}")
    
    # 2. Análisis de 'road_length_m' (Longitud de calles dentro de la manzana)
    # Filtramos las manzanas que SI tienen calles (> 0)
    con_calles = df_net[df_net['road_length_m'] > 0]
    pct_cobertura = (len(con_calles) / len(df_net)) * 100
    
    print(f"\nCOBERTURA DE RED VIAL:") # Indicamos las manzanas con calles detectadas
    print(f"Manzanas con calles detectadas: {len(con_calles)} de {len(df_net)} ({pct_cobertura:.2f}%)")
    
    # 3. Estadísticas solo de las manzanas que tienen datos
    if len(con_calles) > 0: # Mostramos estadísticas solo con las manzanas con calles (con datos)
        print("\nEstadísticas (solo manzanas con calles):")
        cols = ['node_count', 'betweenness_mean', 'road_length_m', 'road_density_km2']
        display(con_calles[cols].describe().round(4))
    else:
        print("\nALERTA CRÍTICA: Ninguna manzana tiene calles. Posible error de proyección (CRS).")

except Exception as e:
    print(f"ERROR: Error leyendo metrics_network: {e}")

✔ Tabla cargada exitosamente. Total de manzanas procesadas: 792

COBERTURA DE RED VIAL:
Manzanas con calles detectadas: 299 de 792 (37.75%)

Estadísticas (solo manzanas con calles):


,node_count,betweenness_mean,road_length_m,road_density_km2
count,299.0000,299.0000,299.0000,299.0000
mean,0.5117,0.0004,115.3699,19767.1582
std,1.0630,0.0019,162.2254,22158.2566
min,0.0000,0.0000,0.2447,29.2057
25%,0.0000,0.0000,34.9886,3689.1162
50%,0.0000,0.0000,78.9916,11249.0308
75%,1.0000,0.0000,161.9454,29219.3675
max,9.0000,0.0180,1970.9657,135663.3332


### 2.1. Validación del dataset integrado de manzanas y red vial

Se muestra que se han cargado exitosamente **792 manzanas censales** correspondientes a la comuna de San Joaquín y la unión con los datos censales muestra una integridad referencial completa, ya que no hay geometrías sin datos.

Al analizar la cobertura de la red, se observa que **299 de las 792 manzanas (37.75%)** tienen métricas viales asociadas (`node_count`, `road_density`).

**Interpretación técnica:** Este porcentaje podría parecer bajo, pero es esperable debido a la metodología de la intersección espacial:
1.  **Metodología:** El script asigna métricas viales solo si los nodos/aristas de la red intersectan o están contenidos en el polígono de la manzana.
2.  **Morfología de San Joaquín:** Muchas manzanas están delimitadas por ejes viales (las calles son los bordes), por lo que geométricamente la calle no está "dentro" de la manzana, sino en el perímetro de estas.
3.  **Zonas industriales:** San Joaquín posee grandes recintos cerrados (como el Campus San Joaquín UC) donde la densidad de vías públicas internas es nula o muy baja en comparación a zonas residenciales densas.

**Validación de Métricas:**
* **Densidad Promedio (~19.7 km/km²):** Es un valor alto, consistente con una zona urbana consolidada y con buena conectividad.
* **Betweenness:** La existencia de valores máximos de `0.018` indica que se han capturado ejes estructurantes importantes dentro de la red comunal.

In [8]:
# --- VALIDACIÓN 3: FUENTES RASTER (IMÁGENES Y TOPOGRAFÍA) ---
# Verifica que las imágenes satelitales (Sentinel-2) y modelos de elevación (DEM) hayan sido registrados correctamente en la base de datos con sus dimensiones espaciales.
print("\n--- Auditoría de Contenido: Catálogo Raster ---")
# Consulta a la tabla de metadatos generada
sql_raster = "SELECT * FROM raw_data.raster_catalog"
try:
    df_raster = pd.read_sql(sql_raster, engine) # Cargar catálogo en DataFrame
    print(f"✔ {len(df_raster)} rasters catalogados en la base de datos.")
    # Mostramos columnas clave para verificar resolución (width/height) y tipo
    display(df_raster[['filename', 'width', 'height', 'source_group']])
except Exception as e: # En caso de error leyendo los rasters
    print(f"ERROR: Error leyendo raster_catalog: {e}")


--- Auditoría de Contenido: Catálogo Raster ---
✔ 5 rasters catalogados en la base de datos.


,filename,width,height,source_group
0,sentinel2_B04.tif,293,528,raw
1,sentinel2_B08.tif,293,528,raw
2,srtm_dem.tif,111,178,raw
3,srtm_dem_32719.tif,100,187,raw
4,S34W071.hgt,3601,3601,raw


### 2.2. Análisis de datos raster: Resolución y escala

La auditoría del catálogo raster revela diferencias importantes en la resolución espacial de las fuentes, lo cual condicionará el nivel de detalle de los análisis posteriores:

1.  **Imágenes satelitales (Sentinel-2):**
-  Dimensiones: `293 x 528` píxeles.
-  **Implicancia:** Al tener una resolución de 10 metros/píxel, se cuenta con un alto nivel de detalle para el cálculo del NDVI. Esto permitirá detectar diferencias de vegetación incluso al interior de manzanas grandes o parques urbanos.

2.  **Modelo de elevación (SRTM):**
- Dimensiones: `100 x 187` píxeles (reproyectado).
- **Implicancia:** La resolución nativa es de ~30 metros (1 arco-segundo). Si bien es suficiente para analizar la pendiente general de la comuna, es una malla más gruesa que la de Sentinel-2. Por lo que no podremos detectar micro-relieves a nivel de calle, pero sí la tendencia topográfica general de la cuenca de Santiago en este sector.

**Conclusión:** La disparidad de las resoluciones es esperable y aceptable para la escala de análisis comunal, pero se debe tener en cuenta al cruzar estas variables: el NDVI aportará variabilidad intra-manzana, mientras que la pendiente actuará como una variable de contexto más suave.o más suave.

In [9]:
# --- VALIDACIÓN 5: INTEGRIDAD DE VARIABLES CRÍTICAS ---
# Validación lógica de los atributos alfanuméricos resultantes del cruce espacial
# Se comprueba que las geometrías de las manzanas hayan heredado correctamente:
# 1. Datos censales (Población, Vivienda) -> Requisito: Variables socioeconómicas
# 2. Normativa urbana (Zonificación) -> Requisito: Planificación territorial

print("--- Validación de contenido ---")

# 1. VERIFICACIÓN DE VARIABLES CENSALES (INE)
try: # Consultamos columnas específicas para confirmar el JOIN con datos demográficos
    sql_censo = "SELECT total_personas, total_viviendas, viv_material_irrecuperable FROM processed_data.manzanas_atributos LIMIT 5"
    df_censo = pd.read_sql(sql_censo, engine)
    # Validamos que la columna clave exista y tenga datos
    if not df_censo.empty and 'total_personas' in df_censo.columns:
        print(f"✔ Requisito Censo 2017: Datos socioeconómicos detectados.")
        print(f"   Variables disponibles: total_personas, total_viviendas, materialidad, etc.")
        display(df_censo.head(2))
    else: # En caso de que no se encuentren las columnas de las personas
        print("ALERTA: No se encontraron las columnas de personas.")
except Exception as e:
    print(f"ERROR: Error verificando Censo: {e}")

# 2. VERIFICACIÓN DE USO DE SUELO (MINVU)
try:
    # Consultamos las categorías únicas de zonificación asignadas a las manzanas
    # Nota: 'nom' suele contener el nombre de la zona (ej: Zona Residencial Mixta)
    sql_usos = """
    SELECT DISTINCT nom, zona 
    FROM processed_data.manzanas_uso_suelo 
    WHERE nom IS NOT NULL 
    LIMIT 5
    """
    df_usos = pd.read_sql(sql_usos, engine)
    
    if not df_usos.empty: # En caso de que el uso de suelo no esté vacío
        print(f"✔ Requisito Uso de Suelo: Zonificación detectada.")
        print(f"   Ejemplos de zonas: {df_usos['nom'].tolist()}")
    else:
        print("ADVERTENCIA: La tabla existe pero la columna 'nom' parece vacía.")
        
except Exception as e: # En caso de error al verificar el uso de suelo
    print(f"ERROR: Error verificando Uso de Suelo: {e}")

--- Validación de contenido ---
✔ Requisito Censo 2017: Datos socioeconómicos detectados.
   Variables disponibles: total_personas, total_viviendas, materialidad, etc.


,total_personas,total_viviendas,viv_material_irrecuperable
0,139,42,0
1,183,65,0


✔ Requisito Uso de Suelo: Zonificación detectada.
   Ejemplos de zonas: ['Zona de Equipamiento Deportivo', 'Faja de protección línea alta tensión y parque', 'Zona Residencial Mixta de Densidad Media', 'Zona Residencial Mixta de Densidad Media-Alta', 'Zona de Infraestructura Energética']


### 2.3. Validación semántica: censo y planificación territorial

Esta etapa confirma que el *dataset integrado* no es solo geométrico, sino que posee otros tipos de atributos importantes para el análisis posterior.

**1. Datos censales (INE 2017):**
La inspección de las primeras filas muestra la presencia de variables demográficas (`total_personas`) y de vivienda (`total_viviendas`), lo que permitirá realizar análisis de segregación y vulnerabilidad en los siguientes notebooks.

**2. Uso de suelo (IDE Minvu):**
Se han detectado categorías de zonificación coherentes con la realidad de San Joaquín:
* **Zona residencial mixta:** Refleja la naturaleza habitacional histórica de la comuna (ej. Población La Legua, El Pinar).
* **Equipamiento deportivo / parques:** Probablemente asociado a zonas como el Parque La Castrina o ejes verdes.
* **Infraestructura energética:** Que concuerda con el paso de líneas de alta tensión o subestaciones presentes en el sector sur-poniente de la capital.

La presencia de estas categorías valida que el cruce espacial (`spatial join`) entre la capa de uso de suelo y las manzanas censales se ejecutó correctamente.orrectamente.

## Conclusiones de la etapa de adquisición

El proceso de descarga y pre-procesamiento de datos para la comuna de **San Joaquín** ha finalizado correctamente, y a partir de las validaciones realizadas, se concluye lo siguiente:

1.  **Completitud respecto a las fuentes solicitadas:** Se han obtenido exitosamente las 7 fuentes de datos mínimas requeridas por el enunciado (Límites, Manzanas, DEM, Sentinel-2, Red Vial, Censo y Uso de Suelo).
2.  **Integridad espacial:** Todos los datos han sido estandarizados al sistema de proyección **EPSG:32719 (WGS 84 / UTM zone 19S)**, lo que garantiza cálculos de áreas y distancias precisos para la zona central de Chile.
3.  **Disponibilidad para Análisis:**
    * La base de datos PostGIS (`geodatabase`) ha sido poblada con las tablas depuradas.
    * Se cuenta con un **Dataset Integrado** (`manzanas_atributos`) que fusiona geometría, demografía y métricas urbanas.